training/network-partition-troubleshooting.md

```diff
@@ -15,3 +15,175 @@ In this lab, you'll... You'll then troubleshoot and resolve the problem.
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ docker exec -it roach-0 /cockroach/cockroach init --insecure
+    ~~~~
+
+4. Verify that the cluster is working by opening the Admin UI at <a href="http://localhost:8080" data-proofer-ignore>http://localhost:8080</a>.
+
+## Step 2. Create a partition in the network
+
+1. Disconnect the nodes in `dc-2` from the shared network backbone:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ docker network disconnect cockroachdbtraining_shared roach-4
+    ~~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ docker network disconnect cockroachdbtraining_shared roach-5
+    ~~~~
+
+## Step 3. Troubleshoot the problem
+
+1. The Admin UI should now show that 2 of the nodes in the cluster have changed from "Healthy" to "Suspect":
+
+    <img src="{{ 'images/training-20.png' | relative_url }}" alt="CockroachDB Admin UI" style="border:1px solid #eee;max-width:100%" />
+
+2. Check whether the "Suspect" nodes are still running by hitting their `/health` endpoints:
+
+    ~~~ shell
+    $ curl localhost:8085/health
+    ~~~~
+
+    ~~~
+    {
+      "nodeId": 4,
+      "address": {
+        "networkField": "tcp",
+        "addressField": "e2f31c1f1e80:26257"
+      },
+      "buildInfo": {
+        "goVersion": "go1.8.3",
+        "tag": "v1.1.5",
+        "time": "2018/02/05 17:40:57",
+        "revision": "ec52ded95a5235d71f26b6cbc59924bd6733fb3a",
+        "cgoCompiler": "gcc 6.3.0",
+        "platform": "linux amd64",
+        "distribution": "CCL",
+        "type": "release-gnu"
+      }
+    ~~~
+
+3. Check whether the "Suspect" nodes consider themselves live by hitting their
+   `/_admin/v1/health` endpoints:
+
+    ~~~ shell
+    $ curl localhost:8085/_admin/v1/health
+    ~~~~
+
+    ~~~
+    {
+      "error": "node is not live",
+      "code": 14
+    }
+    ~~~
+
+4. Check the logs of the downed nodes to see if they contain any clues, where you should find errors like "Error while dialing", "no such host", and "the connection is unavailable":
+
+    ~~~ shell
+    $ docker logs roach-5
+    ~~~~
+
+    ~~~
+    ...
+    I180213 20:38:14.728914 211 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp: lookup roach-1 on 127.0.0.11:53: no such host"; Reconnecting to {roach-1:26257 <nil>}
+    ...
+    W180213 20:38:20.093309 286 storage/node_liveness.go:342  [n5,hb] failed node liveness heartbeat: failed to send RPC: sending to all 3 replicas failed; last error: {<nil> rpc error: code = Unavailable desc = grpc: the connection is unavailable}
+    ~~~
+
+5. Check whether the majority nodes are able to talk to the minority nodes at all by looking at the network latency debug page at <a href="http://localhost:8080/#/reports/network" data-proofer-ignore>http://localhost:8080/#/reports/network</a>
```

Comment:
nit: Missing a colon at end of the sentence.

Reply:
Done.